### PR TITLE
Add validation for system cluster name for both controller and cli

### PR DESCRIPTION
**Pull request by galal-hussein**
- https://github.com/rancher/k3k/issues/69

### CODE REVIEW

File: cli/cmds/cluster/create.go [hunk 1]
```diff
@@ -210,6 +210,9 @@ func validateCreateFlags(clx *cli.Context) error {
 	if name == "" {
 		return errors.New("empty cluster name")
 	}
+	if name == "system" {
```
Comment (briandowns) [inline cli/cmds/cluster/create.go:213]:
```suggestion
	if name == clusterInvalidName {
```

File: pkg/controller/cluster/cluster.go [hunk 2]
```diff
@@ -178,6 +179,11 @@ func (c *ClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request
 }
 
 func (c *ClusterReconciler) createCluster(ctx context.Context, cluster *v1alpha1.Cluster) error {
+
```
Comment (briandowns) [inline pkg/controller/cluster/cluster.go:182]:
```suggestion
```

File: pkg/controller/cluster/cluster.go [hunk 3]
```diff
@@ -178,6 +179,11 @@ func (c *ClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request
...
+	if cluster.Name == clusterInvalidName {
+		klog.Errorf("Invalid cluster name %s, no action will be taken", cluster.Name)
```
Comment (briandowns) [inline pkg/controller/cluster/cluster.go:184]: It's fine to return nil here?

Comment (galal-hussein) [reply pkg/controller/cluster/cluster.go:184]: yes I don't want the controller to reconcile again, just show an error and stop processing this cluster object, in the future, we need to add the status of the cluster in the cluster object as well

**Review by briandowns — APPROVED** [review]

**Pull request merged**